More rules and matching tests

savili opened the pull request:
The solomagicS and testS files each contain new rules in addition to the ones in the respective preceding files. I realise (now) that that wasn't so clever. Sorry!
The whole Appl(e) thing is a bit bulky. The trouble is that if I were to use regex to change all words 
ending in -no or -ndi, for example, they'd have to be special replacement rules, because there are also
entries like 'korino' and 'vundi' in the dictionary, which I wouldn't want to change into 'koringno' or 'vungri'. So I have to check the quantity and frequency of such words first, before I can decide if regex would be feasible here...

savili committed:
Add more replacement rules

I have made four more simple replacement rules and matching tests: Gaha,
Osong, Samang, and  Vitu
- The rules are in solomagicS1.py
- There is matching testS11.py
- The test .in/out files are in MoreReplacementTests

savili committed:
Add one more Xta rule

This one contains parentheses, but no regex. I hope it'll work as is
Please check the order of rules in testS12.py - I don't know how to do
it well

savili committed:
Add four eating rules

Four more rules in solomagicS3.py with matching tests in EatingSuite and
test_eat in testS13.py
Please note the excellent file naming instance

savili committed:
Add Appl(e) rules and tests

13 new rules
I'm not sure yet if it'll be feasible to summarise the 3p-rules by a
regex, so I'll leave it as is for now (I want to check first how many
items I have ending in -ndi that no Appl(e) rule should ever apply
to...)
- AppleSuite contains the test in/out files
- solomagicS4.py contains all the new rules
- There's a new test_apple in testS14.py

aleixpol commented:
Why do you create these S1..S4 files?
Do it on the actual files (solomagic.py) git will take care of making sure we keep track of the different steps.

savili committed:
Add new solomagic and test

These are for the rules Gaha, Osong, Samang, and Vitu

savili committed:
Add new solomagic and test

These are for qutaToXta

savili committed:
Add new solomagic and test

These are for test_eat.
The rules are EatThem, ItsEat, TheirEat, and TheirEatR

savili committed:
Add new solomagic and test

These are for test_apple.
The rules are ApplAtae3p, ApplEnaqa3p, ApplI1pe, ApplI1s, ApplI2s,
ApplI3p, ApplI3s, ApplLagali3p, ApplPoro3p, ApplSali1s, ApplSali2s,
ApplSali3p, and ApplHatoiva3p
